src/v/tristate.h

```diff
@@ -61,6 +69,12 @@ class tristate {
         return is_disabled() || get_optional().has_value();
     }
 
+    /// \brief Checks if the tristate is in the not "Disabled" and not "Set"
+    /// states. we usually use this state to means that the values should be
+    /// retrieved from some other sources, like cluster default values for topic
+    /// properties. the logic is !is_engaged, so this is a convenience method
+    constexpr bool is_empty() const { return !is_engaged(); }
```

Comment:
nit: Isn't this equivalent to == std::nullopt? Might be simpler to define this and `is_engaged()` with comparison

Reply:
if i understand you correctly, you mean being able to write

```cpp
tristate<size_t> val=...
val == std::nullopt; // true iff not disabled and not set
val == disable_tristate; // true iff in disabled state
```

if yes that would be cool but I'll leave it for a followup hoping for a green ci here 🤞

Reply:
Something like that, or just defining these as `*this == tristate<T>{nullopt}`, for example. Especially when it comes to reading the impl, it feels easier to reason about the underlying values, even if we don't necessarily want to expose the values publicly to avoid missteps

Reply:
ah that is a great suggestion